lib/man_task.rb

```diff
@@ -16,12 +16,57 @@
 # you may find current contact information at www.suse.com
 
 class ManTask
-  def man
+  def self.compile_documentation
+    docs = Inspector.all_scopes.map do |scope|
+      scope_doc = "* #{scope}\n\n"
+      scope_doc += YAML.load_file(
+        File.join(Machinery::ROOT, "plugins/#{scope}/#{scope}.yml")
+      )[:description]
+      scope_doc + "\n"
+    end.join
+    File.write("manual/docs/machinery_main_scopes.1.md", docs)
+    Dir.chdir(File.join(Machinery::ROOT, "manual")) do
+      Cheetah.run("mkdocs", "build")
+    end
+  end
+
+  def man(options)
+    if options[:html]
+      man_html(options)
+    else
+      man_system
+    end
+  end
+
+  def man_system
     LocalSystem.validate_existence_of_package("man")
-    system("man", man_path)
+    system("man", File.join(Machinery::ROOT, "man/generated/machinery.1.gz"))
   end
 
-  def man_path
-    File.join(Machinery::ROOT, "man/generated/machinery.1.gz")
+  def man_html(options)
+    if !File.exists?(File.join(Machinery::ROOT, "manual/site"))
```

Comment:
`File.exists?` is deprecated in favor of `File.exist?`.<br>Favor `unless` over `if` for negative conditions.